src/functions/updateDmsTaskWithDates/framework/task-modifier.ts

```diff
@@ -24,33 +25,37 @@ export const modifyTask = async (): Promise<void> => {
 
   const dateTaskName = `${environmentPrefix}-${dateFilteredTaskName}`;
 
-  await stopTaskIfExistsAndRunning(dateTaskName, dms, logger);
+  const taskStatus = await stopTaskIfExistsAndRunning(dateTaskName, dms, logger);
 
-  await dms.createOrModifyTask(dateTaskName,
-                               replicationInstanceArn,
-                               sourceArn,
-                               targetArn,
-                               addDateFilters);
+  // prevent function from being called if already modifying
+  if (taskStatus !== TaskStatus.MODIFYING) {
+    await dms.createOrModifyTask(dateTaskName,
+                                 replicationInstanceArn,
+                                 sourceArn,
+                                 targetArn,
+                                 addDateFilters);
+  }
 
   await startTaskWhenReady(dateTaskName, dms, logger);
 };
 
 async function startTaskWhenReady(taskName: string, dms: DmsApi, logger: ILogger) {
-  await dms.waitForDesiredTaskStatus(taskName, ['ready', 'stopped']);
+  await dms.waitForDesiredTaskStatus(taskName, [TaskStatus.READY, TaskStatus.STOPPED]);
   const startStatus = await dms.startTask(taskName, 'reload-target');
   logger.debug(`status of startTask is ${startStatus}`);
 
 }
-async function stopTaskIfExistsAndRunning(taskName: string, dms: DmsApi, logger: ILogger) {
+async function stopTaskIfExistsAndRunning(taskName: string, dms: DmsApi, logger: ILogger): Promise<string> {
   let taskStatus = '';
   try {
     taskStatus = await dms.getTaskStatus(taskName);
+    logger.debug(`Current status of ${taskName} is ${taskStatus}`);
   } catch (error) {
     logger.debug(`taskname ${taskName} doesn't exist`);
-    taskStatus = 'nonexistant';
```

Comment:
Hmm the old spelling was `nonexistant` which is not grammatically correct, could this be the cause of the issue? i.e. AWS dms returned `nonexistent` and therefore the code was not handling the response correctly?

Reply:
this is a local only status. if all instances match its correct.  I felt safer with a enum and then added a different spelling error :D